### ISSUE TITLE
Multiple issues with edge editing in network

### ISSUE THREAD

**Issue by mitya57** [almende/vis#489]
In all examples below, I am creating an empty `vis.Network` with data manipulation enabled, and populate it using the edit toolbar.
#### Test Case 1

TL;DR: when moving edge, its ownership information is not updated.
- Create three nodes (A, B, C).
- Draw an edge between A and B.
- Select that edge and click Edit.
- Move one of the edge ends from A to C, so that the edge connects B and C.
- Click Back.
- Delete node A.

**Expected result**: node between B and C is not deleted.

**Got instead**: it is deleted.
#### Test Case 2

TL;DR: one of edge ends can be made bound to nothing.
- Create two nodes and draw an edge between them.
- Select that edge and click Edit.
- Move one of the edge ends to the empty space.
- Click Back.

**Expected result**: it should not be possible to do that.

**Got instead**: now one of the nodes is connected with a point in empty space (it behaves as if we hummered a nail at the end of the edge).
#### Test Case 3

TL;DR: you can make stuff flying around.
- Create two nodes and draw an edge between them.
- Select that edge and click Edit.
- Move one of the ends to the empty space.
- Move another end to the empty space.

**Expected result**: it should not be possible to do that.

**Got instead**: the edge **flies away**. Depending on how the disconnect was done, it can be flying away and sometimes returning and flying by like a bee (though without buzzing), or just flying around in a cyclic orbit.

---

I have recorded [a video of test case 2](http://mitya57.me/curiosa/visjs.ogv) with three nodes. Watch it, it is quite funny :)
